matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/sync/handler/SyncResponsePostTreatmentAggregatorHandler.kt

```diff
@@ -68,4 +76,21 @@ internal class SyncResponsePostTreatmentAggregatorHandler @Inject constructor(
             updateUserAccountDataTask.execute(UpdateUserAccountDataTask.DirectChatParams(directMessages = directChats))
         }
     }
+
+    private suspend fun fetchAndUpdateUsers(usersToFetch: List<String>) {
+        val userProfiles = fetchUsers(usersToFetch)
+        userProfiles.forEach { saveUserLocally(monarchy, it) }
+    }
+
+    private suspend fun fetchUsers(usersToFetch: List<String>) = usersToFetch.map {
+        val profileJson = profileService.getProfile(it)
```

Comment:
What will happen if the request fails for any reason, such as no network? I think it will crash the app. Maybe protect by using `tryOrNull`. We will just ignore updating user in that case, this is not a big deal I think.

Reply:
A very good point